src/erlfmt.erl

```diff
@@ -64,18 +67,42 @@ init(State) ->
     rebar3_fmt_prv:init(State).
 
 %% API entry point
--spec format_file(file:name_all(), out()) -> {ok, [error_info()]} | {error, error_info()}.
-format_file(FileName, Out) ->
+-spec format_file(file:name_all(), config()) -> {ok, [error_info()]} | {error, error_info()}.
+format_file(FileName, {RequirePragma, Out}) ->
     try
         {ok, Forms, Warnings} = file_read_forms(FileName),
-        [$\n | Formatted] = format_forms(Forms),
-        verify_forms(FileName, Forms, Formatted),
-        write_forms(FileName, Formatted, Out),
-        {ok, Warnings}
+        ShouldFormat = not RequirePragma or contains_pragma_forms(Forms),
+        case ShouldFormat of
+            true -> [$\n | Formatted] = format_forms(Forms),
+                    verify_forms(FileName, Forms, Formatted),
+                    write_forms(FileName, Formatted, Out),
+                    {ok, Warnings};
+            false -> {ok, Warnings}
+        end
     catch
         {error, Error} -> {error, Error}
     end.
 
+-spec contains_pragma_forms([erlfmt_parse:abstract_form()]) -> boolean().
+contains_pragma_forms(Forms) -> lists:any(fun contains_pragma_form/1, Forms).
+
+contains_pragma_form(Form) ->
+    case Form of
+        {attribute, Meta, _AfAtom, _AbstractExprs} ->
+            {PreComments, PostComments} = erlfmt_format:comments(Meta),
+            lists:any(fun contains_pragma_comment/1, PreComments ++ PostComments);
+        _ -> false
+        end.
+
+contains_pragma_comment({comment, _Loc, Comments}) ->
+    lists:any(fun (Comment) ->
+        case string:find(Comment, "@format") of
+            nomatch -> false;
+            _ -> true
+        end
+    end, Comments);
```

Comment:
A list of strings is also a string, so this can be simplified to 
```suggestion
    string:find(Comments, "@format") =/= nomatch;
```

Reply:
Wait what?
`string:find` can search a list of comments?
Oh because a list of strings, is just a list of characters and a list of list of characters, is flattened like an IO list, when finding?

Reply:
Yeah, it's because of a way `string` module defines strings as `unicode:chardata()` which expands to something like:
```erl
chardata() :: unicode_binary() | charlist().
charlist() :: maybe_improper_list(char() | unicode_binary() | charlist(), unicode_binary() | []).
```

Reply:
Who needs monads when you have the type doing the flattening for you :)

```
result(v, e) :: ok string | error(e)
error(e) :: maybe_improper_list(e)
```

Thank you this is really good to know